### PR TITLE
zsh-clipboard: init at 1.0

### PR DESCRIPTION
**Pull request by bb2020**
###### Motivation for this change

A simple `ohmyzsh` plugin for `zsh` clipboard integration. I've put source code in the package itself, I hope this is allowed.
It can be installed like this:
```
programs.zsh.ohMyZsh = {
  customPkgs = [ pkgs.zsh-clipboard ];
  plugins = [ "clipboard" ];
};
```

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

**Comment by r-rmcgibbo** [comment]
Result of `nixpkgs-review pr 115450` at 2951f6bc run on aarch64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li><a href="https://gist.github.com/a7f76654c9377c9aa855bce35cc1f9c8">zsh-clipboard</a></li></li>
  </ul>
</details>
<details>
  <summary>1 suggestion:</summary>
  <ul>
    <li>warning: <a href="https://github.com/jtojnar/nixpkgs-hammering/blob/master/explanations/missing-phase-hooks.md">missing-phase-hooks</a>

`installPhase` should probably contain `runHook preInstall` and `runHook postInstall`.

Near pkgs/shells/zsh/zsh-clipboard/default.nix:11:3:
```
   |
11 |   installPhase = ''
   |   ^
```
</li>
  </ul>
</details>

File: pkgs/shells/zsh/zsh-clipboard/default.nix [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+  meta = with lib; {
+    description = ''
+      Ohmyzsh plugin that integrates kill-ring with system clipboard.
+
+      Key bindings for C-y, C-k, C-u, M-d, M-backspace and M-w are rebound.
+      Behaviour of these keys should not be changed.
+    '';
+    license = licenses.mit;
+    maintainers = with maintainers; [ bb2020 ];
+    platforms = with platforms; (linux ++ darwin);
```
Comment (SuperSandro2000) [inline pkgs/shells/zsh/zsh-clipboard/default.nix:24]:
```suggestion
    platforms = platforms.unix;
```

File: pkgs/shells/zsh/zsh-clipboard/default.nix [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+    install -D -m0444 -t $out/share/zsh/plugins/clipboard ./clipboard.plugin.zsh
+  '';
+
+  meta = with lib; {
+    description = ''
+      Ohmyzsh plugin that integrates kill-ring with system clipboard.
+
+      Key bindings for C-y, C-k, C-u, M-d, M-backspace and M-w are rebound.
+      Behaviour of these keys should not be changed.
+    '';
```
Comment (SuperSandro2000) [inline pkgs/shells/zsh/zsh-clipboard/default.nix:21]:
```suggestion
    description = "Ohmyzsh plugin that integrates kill-ring with system clipboard";
    longDescription = ''
      Ohmyzsh plugin that integrates kill-ring with system clipboard.

      Key bindings for C-y, C-k, C-u, M-d, M-backspace and M-w are rebound.
      Behaviour of these keys should not be changed.
    '';
```

**Comment by bb2020** [comment]
Thanks for the quick review.

File: pkgs/shells/zsh/zsh-clipboard/default.nix [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+    description = "Ohmyzsh plugin that integrates kill-ring with system clipboard";
+    longDescription = ''
+      Ohmyzsh plugin that integrates kill-ring with system clipboard.
+
+      Key bindings for C-y, C-k, C-u, M-d, M-backspace and M-w are rebound.
+      Behaviour of these keys should not be changed.
+    '';
+    license = licenses.mit;
+    maintainers = with maintainers; [ bb2020 ];
+    platforms = with platforms; unix;
```
Comment (SuperSandro2000) [inline pkgs/shells/zsh/zsh-clipboard/default.nix:25]:
```suggestion
    platforms = platforms.unix;
```

**Comment by SuperSandro2000** [comment]
@ofborg eval

**Comment by bb2020** [comment]
Thanks for merging.

**Pull request merged**